packages/console/src/pages/SignInExperience/components/BrandingForm.tsx

```diff
@@ -0,0 +1,72 @@
+import { BrandingStyle, SignInExperience } from '@logto/schemas';
+import React from 'react';
+import { Control, Controller, UseFormRegister, UseFormWatch } from 'react-hook-form';
+import { useTranslation } from 'react-i18next';
+
+import FormField from '@/components/FormField';
+import RadioGroup, { Radio } from '@/components/RadioGroup';
+import Switch from '@/components/Switch';
+import TextInput from '@/components/TextInput';
+
+import * as styles from './index.module.scss';
+
+type Props = {
+  register: UseFormRegister<SignInExperience>;
+  control: Control<SignInExperience>;
+  watch: UseFormWatch<SignInExperience>;
+};
+
+const BrandingForm = ({ register, control, watch }: Props) => {
+  const { t } = useTranslation(undefined, { keyPrefix: 'admin_console' });
+
+  const isDarkModeEnabled = watch('branding.isDarkModeEnabled');
+  const style = watch('branding.style');
```

Comment:
Use `getValues` instead of `watch` as much as you can, because watch will most likely trigger re-renders
<img width="900" alt="image" src="https://user-images.githubusercontent.com/12833674/163332261-2d375368-4adb-4caf-9c60-5683dc55503d.png">

Reply:
We need to trigger re-render here, to refresh the "required" label in form fields.